RequestManagementSystem/Client/ReqClient.py

```diff
@@ -379,10 +391,11 @@ def readRequestsForJobs(self, jobIDs):
       return readReqsForJobs
     ret = readReqsForJobs["Value"]
     # # create Requests out of JSONs for successful reads
-    if "Successful" in ret:
-      for jobID, fromJSON in ret["Successful"].items():
-        ret["Successful"][jobID] = Request(fromJSON)
-    return S_OK(ret)
+    # Do not forget to cast back str keys to int
+    successful = dict((int(jobID), Request(jsonReq)) for jobID, jsonReq in ret['Successful'].iteritems())
```

Comment:
Change to dictionary comprehension and does it need to use `iteritems`?
```suggestion
    successful = {int(jobID): Request(jsonReq) for jobID, jsonReq in ret['Successful'].items()}
```